### PR TITLE
Added ball_triangle_path_colored

### PR DESCRIPTION
**Pull request by woshikie**
My client liked the ball_triangle_path loading, but wanted to have it with different color.
This commit adds just that, with minimal changes to other files.

### CODE REVIEW

**Comment by woshikie** [comment]
Added #3  as alternative for the same functionality

File: lib/src/shape/indicator_painter.dart [hunk 1]
```diff
@@ -175,8 +165,5 @@ class _ShapePainter extends CustomPainter {
   }
 
   @override
-  bool shouldRepaint(_ShapePainter oldDelegate) =>
-      this.shape != oldDelegate.shape ||
-      this.color != oldDelegate.color ||
-      this.data != oldDelegate.data;
+  bool shouldRepaint(_ShapePainter oldDelegate) => this.shape != oldDelegate.shape || this.color != oldDelegate.color || this.data != oldDelegate.data;
```
Comment (TinoGuo) [inline lib/src/shape/indicator_painter.dart:168]: missed the index check

File: lib/src/decorate/decorate.dart [hunk 2]
```diff
@@ -6,15 +6,15 @@ import 'package:loading_indicator/loading_indicator.dart';
...
-  const DecorateData({@required this.indicator, this.color = Colors.white});
+  const DecorateData({@required this.indicator, this.color = Colors.white, this.colors});
 
   @override
   bool operator ==(other) {
     if (other.runtimeType != runtimeType) return false;
     final DecorateData typedOther = other;
-    return this.color == typedOther.color &&
-        this.indicator == typedOther.indicator;
+    return this.color == typedOther.color && this.indicator == typedOther.indicator;
```
[the remark this answers is not in the document]
Comment (woshikie) [reply lib/src/decorate/decorate.dart]: we can change it to 
```  
  @override
  bool operator ==(other) {
    if (other.runtimeType != runtimeType) return false;
    final DecorateData typedOther = other;
    return this.color == typedOther.color &&
        this.colors == typedOther.colors &&
        this.indicator == typedOther.indicator;
  }
```

or we can change it to a more simple check 
```  
  @override
  bool operator ==(other) {
    if (other.runtimeType != runtimeType) return false;
    return identical(this, other);
  }
```

File: lib/src/decorate/decorate.dart [hunk 3]
```diff
@@ -37,8 +37,7 @@ class DecorateContext extends InheritedWidget {
   }) : super(key: key, child: child);
 
   @override
-  bool updateShouldNotify(DecorateContext oldWidget) =>
-      oldWidget.decorateData == this.decorateData;
+  bool updateShouldNotify(DecorateContext oldWidget) => oldWidget.decorateData == this.decorateData;
```
[the remark this answers is not in the document]
Comment (woshikie) [reply lib/src/decorate/decorate.dart]: Reformatted with the line break at 80

File: lib/src/shape/indicator_painter.dart (hunk 1 above)
Comment (woshikie) [reply lib/src/shape/indicator_painter.dart]: It's not needed here

Comment (woshikie) [reply lib/src/shape/indicator_painter.dart]: I don't think it's needed here

**Comment by woshikie** [comment]
Hello, I've reformatted it with linebreak at 80.

File: lib/src/decorate/decorate.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (TinoGuo) [reply lib/src/decorate/decorate.dart]: pls do not use `identical` here, as we don't compare the address of the object, we compare the fields of the object. pls follow the first approach.

File: lib/src/decorate/decorate.dart (hunk 3 above)
[the remark this answers is not in the document]
Comment (TinoGuo) [reply lib/src/decorate/decorate.dart]: as far as i know, it doesn't follow the dartfmt, pls use `dartfmt -w lib` to format all your files

File: lib/src/shape/indicator_painter.dart (hunk 1 above)
Comment (TinoGuo) [reply lib/src/shape/indicator_painter.dart]: yup, no need, my fault

**Comment by TinoGuo** [comment]
pls rebase the master, added some build check

File: lib/src/decorate/decorate.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (woshikie) [reply lib/src/decorate/decorate.dart]: Will add it in a new commit

File: lib/src/decorate/decorate.dart (hunk 3 above)
[the remark this answers is not in the document]
Comment (woshikie) [reply lib/src/decorate/decorate.dart]: Latest commits have been formatted by dartfmt with default settings.

**Pull request merged**